packages/zealot/src/client/client.ts

```diff
@@ -0,0 +1,229 @@
+import {EventSourcePolyfill} from "event-source-polyfill"
+import nodeFetch from "node-fetch"
+import {PoolConfig, PoolStats, zjson} from ".."
+import {decode} from "../encoder"
+import {ResultStream} from "../query/result-stream"
+import {parseContentType} from "../util/content-type"
+import {createError} from "../util/error"
+import {
+  ClientOpts,
+  CreatePoolOpts,
+  CreatePoolResp,
+  CrossFetch,
+  LoadOpts,
+  Pool,
+  QueryOpts,
+  ResponseFormat
+} from "./types"
+
+export class Client {
+  public fetch: CrossFetch
+  public auth: string | null
+
+  constructor(public baseURL: string, opts: Partial<ClientOpts> = {}) {
+    const defaults: ClientOpts = {env: getEnv(), auth: null}
+    const options: ClientOpts = {...defaults, ...opts}
+    this.auth = options.auth || null
+    this.fetch = options.env === "node" ? nodeFetch : window.fetch.bind(window)
+  }
+
+  async version() {
+    const resp = await this.fetch(this.baseURL + "/version")
+    const content = await parseContentType(resp)
+    return resp.ok ? content : Promise.reject(createError(content))
+  }
+
+  async authMethod() {
+    // TODO
+    return Promise.resolve({} as any)
+  }
+
+  async query(query: string, opts: Partial<QueryOpts> = {}) {
+    const defaults: QueryOpts = {format: "zjson", controlMessages: true}
+    const options: QueryOpts = {...defaults, ...opts}
+    // Wrap the supplied abort with our own
+    const abort = new AbortController()
+    options.signal?.addEventListener("abort", () => {
+      abort.abort()
+    })
+
+    const resp = await this.fetch(this.baseURL + "/query", {
+      method: "POST",
+      body: JSON.stringify({query}),
+      // consolodate the creating of these fetch arguments
+      headers: {
+        Accept: getAcceptValue(options.format),
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${this.auth}`
```

Comment:
This is the only place I'm seeing us use auth and it looks like if you don't provide auth while instantiating the client then it will still set an Authorization header to `Bearer undefined`. Better to not set the header at all imo, and we def will want to include auth on most of the other endpoints (not needed for version, and authMethod)

Reply:
Yes, true. I'll need to do an auth pass on the client before merging this.